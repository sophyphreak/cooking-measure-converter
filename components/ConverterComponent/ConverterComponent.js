import { Button, Col, Row, Table } from 'reactstrap'
import Link from 'next/link'

import Mass from '../Mass/Mass'
import Volume from '../Volume/Volume'
import Temperature from '../Temperature/Temperature'
import Length from '../Length/Length'

const ConverterComponent = ({
  massProps,
  volumeProps,
  temperatureProps,
  lengthProps
}) => (
  <Row className="main animated fadeIn">
    <Col xs="12" sm={{ size: 4, offset: 4 }}>
      <h1 className="main--title">Cooking Measure Converter</h1>
      <Table>
        <thead>
          <tr align="center">
            <th />
            <th>Metric</th>
            <th>Imperial</th>
            <th />
          </tr>
        </thead>
        <tbody>
          <Mass massProps={massProps} />
          <Volume volumeProps={volumeProps} />
          <Length lengthProps={lengthProps} />
          <Temperature temperatureProps={temperatureProps} />
        </tbody>
      </Table>
      <br />
      <br />
      <Link href="/contact">
        <Button className="align-center" color="link" size="sm">
          Contact us
        </Button>
      </Link>
    </Col>
  </Row>
)

export default ConverterComponent
