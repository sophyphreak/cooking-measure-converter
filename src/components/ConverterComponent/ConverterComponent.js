import React from 'react';
import { Col, Row, Table } from 'reactstrap';
import Mass from '../Mass/Mass';

const ConverterComponent = ({ massProps }) => (
  <Row className="main animated fadeIn">
    <Col xs="12" sm={{ size: 4, offset: 4 }}>
      <h1 className="main--title">React Cooking Measure Converter</h1>
      <Table>
        <thead>
          <tr align="center">
            <th />
            <th>Metric</th>
            <th>Imperial</th>
            <th />
          </tr>
        </thead>
        <tbody>
          <Mass massProps={massProps} />
        </tbody>
      </Table>
    </Col>
  </Row>
);

export default ConverterComponent;
