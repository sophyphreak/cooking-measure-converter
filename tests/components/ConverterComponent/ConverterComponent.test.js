import React from 'react';
import { shallow } from 'enzyme';
import ConverterComponent from '../../../src/components/ConverterComponent/ConverterComponent';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<ConverterComponent />);
});

test('should render ConverterComponent correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
