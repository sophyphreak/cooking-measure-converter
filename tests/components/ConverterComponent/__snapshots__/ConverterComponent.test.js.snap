// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConverterComponent correctly 1`] = `
<Row
  className="main animated fadeIn"
  tag="div"
>
  <Col
    sm={
      Object {
        "offset": 4,
        "size": 4,
      }
    }
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="12"
  >
    <h1
      className="main--title"
    >
      React Cooking Measure Converter
    </h1>
    <Table
      responsiveTag="div"
      tag="table"
    >
      <thead>
        <tr
          align="center"
        >
          <th />
          <th>
            Metric
          </th>
          <th>
            Imperial
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <Mass />
        <Volume />
      </tbody>
    </Table>
  </Col>
</Row>
`;
